tests/utils.py

```diff
@@ -161,7 +161,11 @@ def select_resource(self, name, driver=None):
         path = (
             f'//a[contains(text(), "{name}")]/../../' '/input[@name="_selected_action"]'
         )
-        driver.find_element(By.XPATH, path).click()
+
+        element = WebDriverWait(driver, 10).until(
+            EC.presence_of_element_located((By.XPATH, path))
+        )
```

Comment:
Let's assume that the `TimeoutError` is raised by this code. I would update the logic here

```suggestion
    try:
        element = WebDriverWait(driver, 10).until(
            EC.presence_of_element_located((By.XPATH, path))
        )
    except Exception: 
        import ipdb
        ipdb.set_trace()
```

The test execution will halt when the code in except block is executed. Now, inspect the webpage. open the JS console on browser (created by selenium) and look for any errors.